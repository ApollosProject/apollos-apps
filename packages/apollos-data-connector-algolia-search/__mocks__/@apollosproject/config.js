const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    TIMEZONE: 'America/New_York',
  },
  ALGOLIA: {
    APPLICATION_ID: 'some-application-id',
    API_KEY: 'some-api-key',
    CONFIGURATION: {},
  },
});

export default ApolloServer;
