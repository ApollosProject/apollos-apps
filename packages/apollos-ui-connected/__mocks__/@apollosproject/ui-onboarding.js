export const checkOnboardingStatusAndNavigate = () => null;
export const onboardingComplete = () => null;
