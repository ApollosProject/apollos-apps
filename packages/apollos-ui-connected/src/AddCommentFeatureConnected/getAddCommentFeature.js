import gql from 'graphql-tag';
import ApollosConfig from '@apollosproject/config';

export default gql`
  query getAddCommentFeature($featureId: ID!) {
    node(id: $featureId) {
      ...AddCommentFeatureFragment
    }
    currentUser {
      id
      profile {
        id
        photo {
          uri
        }
      }
    }
  }
  ${ApollosConfig.FRAGMENTS.ADD_COMMENT_FEATURE_FRAGMENT}
`;
