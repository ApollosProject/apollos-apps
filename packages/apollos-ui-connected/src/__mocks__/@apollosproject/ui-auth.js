export const ProtectedTouchable = ({ children }) => children;
