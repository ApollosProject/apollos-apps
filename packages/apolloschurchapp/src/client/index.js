import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import SplashScreen from 'react-native-splash-screen';
import gql from 'graphql-tag';

import httpLink from './httpLink';
import clientStateLink from './clientStateLink';
import authLink from './authLink'; // eslint-disable-line
import cache, { ensureCacheHydration } from './cache';

const link = clientStateLink.concat(ApolloLink.from([authLink, httpLink]));

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: true,
  shouldBatch: true,
});

client.onResetStore(clientStateLink.writeDefaults);

export const CACHE_LOADED = gql`
  query {
    cacheLoaded @client
  }
`;

export const MARK_CACHE_LOADED = gql`
  mutation markCacheLoaded {
    cacheMarkLoaded @client
  }
`;

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    try {
      await ensureCacheHydration;
    } catch (e) {
      throw e;
    } finally {
      if (SplashScreen && SplashScreen.hide) SplashScreen.hide();
      client.mutate({ mutation: MARK_CACHE_LOADED });
    }
  }

  render() {
    return <ApolloProvider {...this.props} client={client} />;
  }
}

export default ClientProvider;
