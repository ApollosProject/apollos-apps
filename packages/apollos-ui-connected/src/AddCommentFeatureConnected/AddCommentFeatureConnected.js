import React from 'react';
import { useMutation, useQuery } from '@apollo/client';
import PropTypes from 'prop-types';
import { AddCommentInput } from '@apollosproject/ui-kit';

import GET_NODE_FEATURE_FEED from '../FeaturesFeedConnected/getFeatureFeed';
import GET_NODE_FEATURES from '../NodeFeaturesConnected/getNodeFeatures';
import GET_COMMENT_LIST_FEATURE from '../CommentListFeatureConnected/getCommentListFeature';

import GET_ADD_COMMENT_FEATURE from './getAddCommentFeature';
import ADD_COMMENT from './addComment';

const AddCommentFeatureConnected = ({
  featureId,
  Component,
  isLoading,
  refetchRef,
  onAddComment,
  ...props
}) => {
  const { data, loading, refetch } = useQuery(GET_ADD_COMMENT_FEATURE, {
    variables: {
      featureId,
    },
  });

  if (featureId && refetch && refetchRef)
    refetchRef({ refetch, id: featureId });

  const currentPerson = data?.currentUser;
  const node = data?.node;

  const addCommentUpdate = (cache, { data: { addComment: newComment } }) => {
    // Let's get started! No crashing allowed.
    // This function is brittle and will fail under a variety of circumstances.
    // That's okay for now, because what those "other circumstances" will be and if they occur
    // is pretty undefined at this point. We can revisit this in the future if needed.
    try {
      // Find the parent of this feature. We need to add the new comment to the comment list on this node.
      const parentNode = cache.readQuery({
        query: GET_NODE_FEATURES,
        variables: { nodeId: node.relatedNode.id },
      }).node;

      // Okay, we have our parent. Let's find the feature feed belonging to this parent.
      const myFeatureFeed = cache.readQuery({
        query: GET_NODE_FEATURE_FEED,
        variables: { featureFeedId: parentNode.featureFeed.id },
      }).node;

      // Nice, we have our list of features (just ids and typenames).
      // Let's find the CommentList that should be attached.
      const commentListFeatureId = myFeatureFeed.features.find(
        ({ __typename }) => __typename === 'CommentListFeature'
      ).id;

      // Now that we have our CommentListFeature, let's get the comment list.
      const commentListFeature = cache.readQuery({
        query: GET_COMMENT_LIST_FEATURE,
        variables: { featureId: commentListFeatureId },
      });

      // Finally, let's add the comment we added to that comment list.
      cache.writeQuery({
        query: GET_COMMENT_LIST_FEATURE,
        variables: { featureId: commentListFeatureId },
        data: {
          ...commentListFeature,
          node: {
            ...commentListFeature.node,
            comments: [newComment, ...commentListFeature.node.comments],
          },
        },
      });
    } catch (e) {
      console.warn('Failed to update cache after adding comment', e);
    }
  };

  const [addComment] = useMutation(ADD_COMMENT, {
    update: onAddComment || addCommentUpdate,
  });

  return (
    <Component
      {...node}
      {...props}
      loading={isLoading || loading}
      onSubmit={(text) =>
        addComment({
          variables: {
            text,
            parentId: node.relatedNode.id,
          },
        })
      }
      profile={currentPerson?.profile}
    />
  );
};

AddCommentFeatureConnected.propTypes = {
  Component: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.object,
  ]),
  featureId: PropTypes.string.isRequired,
  isLoading: PropTypes.bool,
  refetchRef: PropTypes.func,
  onAddComment: PropTypes.func,
};

AddCommentFeatureConnected.defaultProps = {
  Component: AddCommentInput,
};

export default AddCommentFeatureConnected;
