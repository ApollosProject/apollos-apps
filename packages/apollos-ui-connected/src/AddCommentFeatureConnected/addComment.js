import { gql } from '@apollo/client';

export default gql`
  mutation addComment(
    $text: String!
    $parentId: ID!
    $visibility: CommentVisibility
  ) {
    addComment(text: $text, parentId: $parentId, visibility: $visibility) {
      id
      text
      person {
        id
        nickName
        photo {
          uri
        }
        campus {
          id
          name
        }
      }
    }
  }
`;
