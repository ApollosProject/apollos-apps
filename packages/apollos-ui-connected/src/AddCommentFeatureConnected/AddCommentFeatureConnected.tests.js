import React from 'react';
import { Providers, renderWithApolloData } from '@apollosproject/ui-test-utils';
import { MockedProvider } from '@apollo/client/testing';

import GET_ADD_COMMENT_FEATURE from './getAddCommentFeature';

import AddCommentFeatureConnected from './AddCommentFeatureConnected';

const mock = {
  request: {
    query: GET_ADD_COMMENT_FEATURE,
    variables: { featureId: 'AddCommentFeature:123' },
  },
  result: {
    data: {
      currentUser: {
        id: 'AuthenticatedUser:123',
        __typename: 'AuthenticatedUser',
        profile: {
          __typename: 'Person',
          id: 'Person:123',
          photo: {
            uri: `https://picsum.photos/200`,
            __typename: 'ImageMediaSource',
          },
        },
      },
      node: {
        id: 'AddCommentFeature:123',
        __typename: 'AddCommentFeature',
        addPrompt: 'A great add prompt',
        initialPrompt: 'A great initial prompt',
        relatedNode: {
          id: 'UniversalContentItem:123',
          __typename: 'UniversalContentItem',
        },
      },
    },
  },
};

describe('The CommentListFeatureConnected component', () => {
  it('should render', async () => {
    const tree = await renderWithApolloData(
      <Providers MockedProvider={MockedProvider} mocks={[mock]}>
        <AddCommentFeatureConnected
          refetchRef={jest.fn()}
          featureId={'AddCommentFeature:123'}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
