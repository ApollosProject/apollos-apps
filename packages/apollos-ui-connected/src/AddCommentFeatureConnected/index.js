import AddCommentFeatureConnected from './AddCommentFeatureConnected';

export ADD_COMMENT from './addComment';
export GET_ADD_COMMENT_FEATURE from './getAddCommentFeature';

export default AddCommentFeatureConnected;
