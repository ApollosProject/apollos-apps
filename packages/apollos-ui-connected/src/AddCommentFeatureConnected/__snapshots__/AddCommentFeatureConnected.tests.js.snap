// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CommentListFeatureConnected component should render 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "borderColor": "rgba(0, 0, 0, 0.1)",
        "borderTopWidth": 0.5,
        "flexDirection": "row",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 48,
          "justifyContent": "center",
          "width": 48,
        }
      }
      themeSize={48}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={52.5}
        bbWidth={52.5}
        height={52.5}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            null,
            Object {
              "flex": 0,
              "height": 52,
              "width": 52,
            },
          ]
        }
        themeSize={48}
        vbHeight={24}
        vbWidth={24}
        width={52.5}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-3.33 8.33C8.67 8.5 10.2 7 12 7s3.33 1.5 3.33 3.33v.84c0 1.84-1.53 3.33-3.33 3.33S8.67 13 8.67 11.17v-.84zm3.33 10c-2.03 0-3.9-.73-5.33-1.93.7-1.32 2.07-2.23 3.66-2.23h3.34c1.6 0 2.97.9 3.66 2.22-1.44 1.2-3.3 1.92-5.33 1.92z"
            fill={
              Array [
                0,
                4289045925,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "bottom": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            borderRadius={24}
            iconPadding={4.8}
            style={
              Object {
                "backgroundColor": "#17B582",
                "padding": 4.8,
                "shadowColor": "rgba(0, 0, 0, 0.09999999999999998)",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 8,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={12}
              bbWidth={12}
              fill="none"
              height={12}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  undefined,
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 12,
                    "width": 12,
                  },
                ]
              }
              vbHeight={10}
              vbWidth={10}
              width={12}
            >
              <RNSVGGroup
                fill={null}
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGPath
                  d="M1.5625 5H8.4375"
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={
                    Array [
                      0,
                      4294967295,
                    ]
                  }
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="2"
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M5 1.5625V8.4375"
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={
                    Array [
                      0,
                      4294967295,
                    ]
                  }
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="2"
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#303030",
          "fontFamily": "InterUI-Medium",
          "fontSize": 14,
          "lineHeight": 21,
          "padding": 8,
        }
      }
    >
      A great initial prompt
    </Text>
  </View>
</View>
`;
